### ISSUE TITLE
Idea: color mixer

### ISSUE THREAD

**Issue by dsizzle** [HaikuArchives/ArtPaint#264]
Disclaimer: I am not a painter

while working on the color palette window I had an idea. I was watching the famous “joy of painting” tv show, and I saw that he mixed all the colors for his picture from a palette of 5 or 6 colors. Digital painting uses RGB or CMYK for instance, one based on the behavior of light and the other based on printing inks. But pigments are different, and so for a paint app what if you could mix colors like a painter?

basically was thinking of color sliders but the user could specify the colors at the ends of the slider. Instead of red, green, blue you could have purple, brown, yellow… or whatever colors you want to mix.

**Comment by humdingerb** [comment]
I'm not a painter either, or much of an artist when it comes to that. Maybe we could use input from @zuMi...
I sounds like an interesting idea anyway and probably not too complicated to implement. It's practically identical to the current color models with colour wells instead of the "R", "G", "B". One could start of with red, green, blue (are 3 colours to mix enough?) and accept colour drops to those slider wells from anywhere.
The "A"lpha slider stays, of course.

While we're on color mixer, I do like Colors! as well. With those H,S,V,R,G,B radio buttons you quickly change the "channel" and pick the color you like.

![colors](https://user-images.githubusercontent.com/6107158/191244753-24ca2ffb-7dd1-4742-b43b-9e4b0a640642.png)

**Comment by dsizzle** [comment]
Colors! looks nearly identical to photoshop’s color picker. I wonder if the author would be ok with incorporating that code into another tool - or making an “official” Haiku BColorPicker control.

but yes, I was thinking either sliders with droppable end colors, or maybe that big square in the Colors! window but you can set any of the corner colors.

**Comment by humdingerb** [comment]
Well, it is MIT licensed, and I bet John would have nothing against that. I think he also has worked a bit on improving BColorPicker some time ago. Let's ask him: Oi, @jscipione, can ArtPaint rip off Colors! ? :)

> or maybe that big square in the Colors! window but you can set any of the corner colors.

That was my thought at first, too. Only I imagined a circle where you could drop any (sane) number of colours on its circumference and move them around. But then I thought to myself, "Nah, you cannot weigh down poor Dale with such a complicated custom interface..." :)
